src/quantforge/backtest_strategy.py

```diff
@@ -0,0 +1,597 @@
+        # This part needs significant updates once metrics calculation is solid.
+        # For now, provide basic info.
+
+        output_lines = []
+        separator = "=" * 50
+
+        output_lines.append("\n" + separator)
+        output_lines.append(f"BACKTEST RESULTS: {self.strategy.name}")
+        output_lines.append(separator)
+        output_lines.append(f"Period: {metrics['start_date'].date()} to {metrics['end_date'].date()}")
+        output_lines.append(f"Initial Capital: ${metrics['initial_capital']:,.2f}")
+        output_lines.append(f"Final Portfolio Value: ${metrics['final_portfolio_value']:,.2f}")
+
+        profit_loss = metrics["final_portfolio_value"] - metrics["initial_capital"]
+        profit_loss_pct = (profit_loss / metrics["initial_capital"] * 100) if metrics["initial_capital"] != 0 else 0
+        output_lines.append(f"Profit/Loss: ${profit_loss:,.2f} ({profit_loss_pct:.2f}%)")
+        output_lines.append(f"Total Return: {metrics.get('total_return', 0.0):.2%}")
+        # Add other available metrics placeholders
+        output_lines.append(f"Number of Trades: {metrics.get('num_trades', 'N/A')}")
+
+
+        output_lines.append("\nSTRATEGY PARAMETERS:")
+        # Assuming strategy params are stored reasonably
+        if hasattr(self.strategy, 'rsi_window'): # Example for RsiStrategy
+             output_lines.append(f"  rsi_window: {self.strategy.rsi_window}")
+             output_lines.append(f"  oversold_threshold: {self.strategy.oversold_threshold}")
+             output_lines.append(f"  overbought_threshold: {self.strategy.overbought_threshold}")
+        # Add more parameter reporting as needed
+
+        output_lines.append("\nFINAL HOLDINGS:")
+        # Use portfolio methods/properties if available
+        open_positions = self.portfolio.get_all_open_positions() # Assuming this method exists or can be added
+        if not open_positions:
+             output_lines.append("  No open positions.")
+        else:
+             # Use the stored current market prices
+             current_prices = self.current_market_prices 
+             for position in open_positions:
+                 item = position.open_transaction.tradeable_item
+                 qty = position.open_transaction.quantity
+                 current_price = current_prices.get(item, 0)
+                 value = qty * current_price
+                 output_lines.append(f"  {item.id}: {qty} shares @ ${current_price:,.2f} (Value: ${value:,.2f})")
+
+
+        output_lines.append(f"\nFinal Cash: ${self.portfolio.cash:,.2f}")
+        output_lines.append(separator)
+
+        # Optional: Add trade summary (Requires portfolio trade tracking)
+        # ...
+
+        return "\n".join(output_lines)
+
+
+    def print_results(self):
+        """Print backtest results and stats."""
+        results_str = self.get_results_as_string()
+        print(results_str)
+
+
+# --- Click CLI (Adapted) ---
```

Comment:
Move all teh main code along with click etc in a backtest runner file. This way we can have many different runners.